packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -1038,15 +1037,11 @@ class AppInstance {
     _logger.close();
   }
 
-  Future<T> _runInZone<T>(AppDomain domain, FutureOr<T> method()) {
-    _logger ??= _AppRunLogger(domain, this, parent: logToStdout ? globals.logger : null);
-
-    return context.run<T>(
-      body: method,
-      overrides: <Type, Generator>{
-        Logger: () => _logger,
-      },
-    );
+  Future<T> _runInZone<T>(AppDomain domain, FutureOr<T> method()) async {
+    final AppRunLogger logger = globals.logger as AppRunLogger;
```

Comment:
I wonder if there's some way we can be more explicit about requiring a `AppRunLogger` context logger whenever a daemon `AppInstance` is used (injecting it into the constructor instead of grabbing out of the context?).  

Seems error-prone that one layer needs to know `AppInstance` is only used for `attach` and `run` (and not `drive`, say) and to set context `AppRunLogger` and then the next layer needs to assume it's been set correctly without enforcing it.

Reply:
The right way to do this is probably to make the Domain require an AppRunLogger in the constructor. We can pull it out of the context and cast in run/attach - at least until those don't use the context.

Though really, we could probably express all of our loggers as a single class (except for maybe BufferLogger) and then it would just be about configuration